src/fixup/fixup.cpp

```diff
@@ -645,6 +664,156 @@ TaskStatus ApplyFloorsImpl(T *rc, IndexDomain domain = IndexDomain::entire) {
   return TaskStatus::complete;
 }
 
+static Real dphi_dt = 0.;
+static Real next_dphi_dt_update_time = 0.;
+static Real phi_factor = 0.;
+
+TaskStatus EndOfStepModify(MeshData<Real> *md, const Real t, const Real dt,
+                           bool last_stage) {
+  auto *pm = md->GetParentPointer();
+  StateDescriptor *fix_pkg = pm->packages.Get("fixup").get();
+
+  const bool enable_phi_enforcement = fix_pkg->Param<bool>("enable_phi_enforcement");
+
+  namespace p = fluid_prim;
+  namespace c = fluid_cons;
+  const std::vector<std::string> vars({p::bfield, c::bfield});
+  PackIndexMap imap;
+  auto pack = md->PackVariables(vars, imap);
+
+  const int pblo = imap[p::bfield].first;
+  const int cblo = imap[c::bfield].first;
+
+  const auto ib = md->GetBoundsI(IndexDomain::interior);
+  const auto jb = md->GetBoundsJ(IndexDomain::interior);
+  const auto kb = md->GetBoundsK(IndexDomain::interior);
+
+  auto geom = Geometry::GetCoordinateSystem(md);
+  auto gpkg = pm->packages.Get("geometry");
+  bool derefine_poles = gpkg->Param<bool>("derefine_poles");
+  Real h = gpkg->Param<Real>("h");
+  Real xt = gpkg->Param<Real>("xt");
+  Real alpha = gpkg->Param<Real>("alpha");
+  Real x0 = gpkg->Param<Real>("x0");
+  Real smooth = gpkg->Param<Real>("smooth");
+  auto tr = Geometry::McKinneyGammieRyan(derefine_poles, h, xt, alpha, x0, smooth);
+
+  if (enable_phi_enforcement) {
+    const Real enforced_phi = fix_pkg->Param<Real>("enforced_phi");
+    const Real enforced_phi_timescale = fix_pkg->Param<Real>("enforced_phi_timescale");
+    const Real enforced_phi_cadence = fix_pkg->Param<Real>("enforced_phi_cadence");
+    const Real enforced_phi_start_time = fix_pkg->Param<Real>("enforced_phi_start_time");
+    if (t > enforced_phi_start_time) {
+
+      // This only needs to be done at initialization
```

Comment:
Could hide it behind a static bool that gets set to true after it's been done once?

Reply:
Yeah that should be fine for restarts (actually I need to switch some of my existing static bools to mutable params (I think) so restarts work with them)

Reply:
Thinking more I'm not sure the `static bool` trick will work here if we have multiple regions per MPI rank.

Reply:
To do this fully correctly (once *per restart*) the thing to do is probably create a mutable parameter for whether the normalization has been evaluated, defaults to false, and is set to true when normalization is evaluated. Then a similar mutable param that stores the normalization parameter. 

Then one *should* also store `dphi_dt` and `next_dphi_dt_update_time` in a parthenon Variable that is carried across restarts. 

Neither of these is necessary to make the thing actually work -- I think the hiccup in `dphi_dt` evaluation cadence during restarts is probably not significant...